### PR TITLE
Fix ConfigInjector metadata generation in config-generator maven plugin

### PR DESCRIPTION
**Pull request by avpinchuk**
Generated `ConfigInjector`'s metadata does not include elements and attributes from an indirectly implemented interfaces.

This fixes it.

**Note**: we should build the GlassFish with this fixed plugin, not with old 2.5.0-b53.

### CODE REVIEW

**Comment by dmatej** [comment]
@arjantijms We need to release the new version of the plugin too.

**Review by dmatej — APPROVED** [review]

**Review by arjantijms — APPROVED** [review]

**Comment by pzygielo** [comment]
> @arjantijms We need to release the new version of the plugin too.

Maybe
- https://github.com/eclipse-ee4j/glassfish/issues/24262

this could be considered as well?

**Comment by avpinchuk** [comment]
@pzygielo, I thought about this. But it will take some time

**Comment by dmatej** [comment]
@arjantijms did not release the plugin for 7.0.4 -> this fix is not inside 7.0.4. What are the consequences for this GlassFish version?

**Comment by avpinchuk** [comment]
No consequences now.

I found a bug while refactoring the config API. Some interfaces annotated `@Configured` implements both `Resource` and `BindableResource`, whiсh implements `Resource` too. I we left only `BindableResource` in the list of implemented interfaces, then we get an error.

To reproduce, remove `Resource` from implemented interfaces in `org.glassfish.connectors.config.AdminObjectResource`. Left only `BindableResource`, then `mvn clean install` the GF. You get an error in unit tests.

**Comment by avpinchuk** [comment]
> @arjantijms did not release the plugin for 7.0.4 -> this fix is not inside 7.0.4. What are the consequences for this GlassFish version?

I want to update the plugin, replace old code, check for the thread-safety etc. What do you think about it?

**Comment by dmatej** [comment]
> > @arjantijms did not release the plugin for 7.0.4 -> this fix is not inside 7.0.4. What are the consequences for this GlassFish version?
> 
> I want to update the plugin, replace old code, check for the thread-safety etc. What do you think about it?

Can you wait for a moment? Just tomorrow. I would try to migrate it to the glassfish build maven plugin. It has a dependency on GF module, but it seems it is just for it's transitive dependencies, so I will try to make it narrow. 
Then next week we could release a next generation of the plugin.

**Comment by avpinchuk** [comment]
> Can you wait for a moment?

No problem. I have a few more bugs in my stash :-)

**Comment by dmatej** [comment]
So, I was wrong. The plugin depends on GlassFish's hk-config classes and GlassFish depends on the plugin. That also explains why we still use the old version of the plugin. I am thinking what we can do with that ...

- Move hk2-config to the refactored gf plugin as it's jar file. It doesn't have any GF dependencies, so it would be safe and correct.
- Move just the plugin goal and make it dependant on older GF version. Bad design.
- Keep it here. Unusable wrong design with recursion.

Basically I can do the first option until the end of this week and we can try it before merging PRs.

**Comment by avpinchuk** [comment]
> So, I was wrong. The plugin depends on GlassFish's hk-config classes and GlassFish depends on the plugin. That also explains why we still use the old version of the plugin. I am thinking what we can do with that ...

At a first glance, we could try to replace all `Class` occurrences with FQN. But it is not exactly. Need further investigation.
